src/Microsoft.AspNetCore.Sockets.Client.Http/ServerSentEventsTransport.cs

```diff
@@ -71,25 +71,25 @@ public Task StartAsync(Uri url, Channel<byte[], SendMessage> application)
             var response = await _httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
 
             var stream = await response.Content.ReadAsStreamAsync();
-
-            var pipelineReader = stream.AsPipelineReader();
+            var pipelineReader = stream.AsPipelineReader(cancellationToken);
+            var readCancellationRegistration = cancellationToken.Register(() => pipelineReader.CancelPendingRead());
```

Comment:
Yea, this is kinda hacky but it's fine. This is fixed in the version os pipelines that kestrel uses.